packages/components/src/components/modal/modal.tsx

```diff
@@ -105,8 +105,9 @@ export class Modal {
   }
 
   public componentDidRender(): void {
-    if (this.open) {
-      // reset scroll top to zero in case content is longer than viewport height, - some timeout is needed although it shouldn't
+    // fix scroll to top bug in react #2569 (useState hook within modal causes rerender of modal)
+    if (this.open && !this.host.contains(document.activeElement)) {
```

Comment:
I think this isn't a good solution since this just fixes the sympton and not the actual cause.  
The real reason for the modals lifecycle and rerender is the new object that is passed to the `fullscreen` prop.